pkg/containerd/cmd/service/prepare.go

```diff
@@ -4,18 +4,79 @@ package main
 // Update it in both places if you make changes
 
 import (
+	"fmt"
+	"os"
 	"path/filepath"
 	"syscall"
 )
 
 func prepare(path string) error {
+	// see if we are dealing with a read only or read write container
+	if _, err := os.Stat(filepath.Join(path, "lower")); err != nil {
```

Comment:
It'd be nice to know about this when diagnosing why ones system isn't booting right though